server/src/main/scala/services/graphstages/ArrivalSplitsGraphStage.scala

```diff
@@ -44,6 +45,14 @@ class ArrivalSplitsGraphStage(name: String = "",
     val log: Logger = LoggerFactory.getLogger(s"$getClass-$name")
 
     override def preStart(): Unit = {
+      optionalInitialManifests match {
+        case Some(manifests) =>
+          log.info(s"Received ${manifests.size} initial manifests")
+          manifestsBuffer = manifests.groupBy(_.key)
+        case None =>
+          log.warn("Did not receive any manifests to initialise with")
+      }
+
```

Comment:
Just a general question with this class. Are we losing state on startup with `arrivalsWithSplitsDiff` and `arrivalsToRemove` or are these states not worth storing/populating?

Reply:
Legit question! Those states are populated when data is received and processed. Ie any incoming data that results in changes to push downstream will populate those two states